include/swift/AST/SubstitutionMap.h

```diff
@@ -99,6 +97,9 @@ class SubstitutionMap {
   SubstitutionMap subst(TypeSubstitutionFn subs,
                         LookupConformanceFn conformances) const;
 
+  /// Build an array of substitutions from a substitution map.
+  void toList(SmallVectorImpl<Substitution> &subs) const;
+
```

Comment:
I think we also want a variant `toList(GenericSignature *, SmallVectorImpl<Substitution> &subs)`. Sometimes you indeed build the map using the original generic signature, but produce a result for the new specialized generic signature which is "compatible", but may have less substitutable generic parameters.

Also, may be there should be a typedef for `SmallVectorImpl<Substitution>`? May be`SusbtitutionListImpl`?